astroid/brain/brain_typing.py

```diff
@@ -351,6 +353,34 @@ def infer_tuple_alias(
     return iter([class_def])
 
 
+def _looks_like_typing_cast(node: Call) -> bool:
+    return isinstance(node, Call) and (
+        isinstance(node.func, Name)
+        and node.func.name == "cast"
+        or isinstance(node.func, Attribute)
+        and node.func.attrname == "cast"
+    )
+
+
+def infer_typing_cast(
+    node: Call, ctx: context.InferenceContext = None
+) -> typing.Union[typing.Iterator[NodeNG], typing.Iterator[Instance]]:
```

Comment:
To fix the tests, you need to add the import for `NodeNG` (from `astroid.node_classes`)